src/node.h

```diff
@@ -617,6 +617,24 @@ NODE_EXTERN v8::MaybeLocal<v8::Value> PrepareStackTraceCallback(
     v8::Local<v8::Value> exception,
     v8::Local<v8::Array> trace);
 
+// Writes a diagnostic report to a file. If filename is not provided, the
+// default filename includes the date, time, PID, and a sequence number.
+// The report's JavaScript stack trace is taken from err, if present.
+// If isolate or env is nullptr, no information about the isolate and env
+// is included in the report.
```

Comment:
We generally don't provide functions in our public API which take both `Isolate*` and `Environment*` arguments, because that is generally redundant. (For internal functions, the story is a bit different, but in the public API we should aim for clarity.)

I'd suggest removing the `env` argument and instead inferring it inside the functions, and, if it makes sense, adding overloads that then take an `Environment*` instead of `Isolate*` for the cases in which users *do* want to explicitly specify an `Environment` instance.